src/scancode/resource.py

```diff
@@ -434,19 +436,45 @@ def create_resources(_seq, _top, _parent, _is_file):
         # not keep parents already walked and we walk topdown.
         parent_by_loc = {root.location: root}
 
+
+        def _walk(max_depth):
+            """Walk through the root_directory recursively upto max_depth nested subdirectories
+            and build the Resource tree. max_depth should be positive or -1 for walking through
+            maximum available nesting levels."""
+
+            # If depth is limited 
+            if max_depth >= 0: 
+                # Note down the directory separator and the depth of root directory
+                # This helps us calculate the depth of nested directories to eliminate
+                # the ones we do not want
+                dir_sep = os.path.sep 
```

Comment:
Why do wee need this `dir_sep` variable when we can refer to `os.path.sep`?

Reply:
Alright. Will get to it, hopefully by tomorrow. Thanks.